app/src/main/cpp/skyline/services/account/IAsyncContext.cpp

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2023 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include "IAsyncContext.h"
+
+namespace skyline::service::account {
+    IAsyncContext::IAsyncContext(const DeviceState &state, ServiceManager &manager) : BaseService(state, manager),
+        systemEvent(std::make_shared<kernel::type::KEvent>(state, true)) {}
+
+    Result IAsyncContext::GetSystemEvent(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        auto handle{state.process->InsertItem(systemEvent)};
+        Logger::Debug("System Event Handle: 0x{:X}", handle);
+
+        response.copyHandles.push_back(handle);
+        return {};
+    }
+
+    Result IAsyncContext::Cancel(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        isComplete.store(true);
+        systemEvent->Signal();
+        return {};
+    }
+
+    Result IAsyncContext::HasDone(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        isComplete.store(true);
+        response.Push<u8>(isComplete.load());
```

Comment:
What's the point of isComplete here, the only possible visible state from the guest is that it's true. Is this a mistake or should it just be dropped